components/insight/SRC/org/openmicroscopy/shoola/agents/util/ui/ScriptingDialog.java

```diff
@@ -490,13 +496,18 @@ public void mouseReleased(MouseEvent e) {
                 }
             } else if (String.class.equals(type)) {
                 if (comp == null) {
-                    comp = new JTextField();
-                    if (defValue != null) {
-                        length = defValue.toString().length();
-                        String s = defValue.toString().trim();
-                        ((JTextField) comp).setColumns(length);
-                        ((JTextField) comp).setText(s);
-                        columnsSet = s.length() > 0;
+                    String nl = name.toLowerCase();
+                    if ("password".equals(nl)) {
+                        comp = new JPasswordField();
+                    } else {
+                        comp = new JTextField();
+                        if (defValue != null) {
+                            length = defValue.toString().length();
```

Comment:
Definitely want the pre-trim length?

Reply:
old code not changed in this PR. 
I could clean it